### ISSUE TITLE
Obs appear to be in umbrella projects when they're not

### ISSUE THREAD

**Issue by kueda** [inaturalist/inaturalist#3237]
**Describe the bug**
When you view https://www.inaturalist.org/observations/138626, [Home Projects Umbrella](https://www.inaturalist.org/projects/home-projects-umbrella) appears under the list of projects, even though the observation doesn't appear to be in any of the projects in that umbrella.

**To Reproduce**
1. Go to https://www.inaturalist.org/observations/138626
1. See that Home Projects Umbrella](https://www.inaturalist.org/projects/home-projects-umbrella) appears under the list of projects in the right column

**Expected behavior**
The obs isn't in any of the umbrella's projects so it shouldn't appear in the umbrella.

**Additional context**
Probably an API bug, but I haven't chased it down yet.

**Comment by carrieseltzer** [comment]
Turns out the issue was somehow related to the inclusion of this project: https://www.inaturalist.org/projects/observations-from-my-home
I just removed that project from the umbrella, and the inappropriate umbrella project badges disappeared.

However, that project's rules are "project members only" _except_ the project creator, but the project creator is the only member of the project. So, it's not clear to me why this project was actually able to trigger the display of the umbrella project badge for everyone who joined the umbrella project when _zero_ observations should have been included in "observations from my home". 

<img width="415" alt="Screen Shot 2021-09-23 at 4 50 15 PM" src="https://user-images.githubusercontent.com/7501183/134582351-37aba633-81e0-42ab-8e4a-4af56e8fbe0d.png">
 
But now, I don't see the umbrella project badge displayed on my observations that _are_ in the umbrella project, e.g. https://www.inaturalist.org/observations/95458203
So now the issue is "obs appear to not be in umbrella projects when they are", which someone else [reported](https://forum.inaturalist.org/t/umbrella-project-shown-on-observations-that-are-not-in-it/26208/6) in the forum thread about the original issue.
